lib/final-api/endpoint/ddtf.rb

```diff
@@ -17,6 +17,43 @@ def registered(app)
             }.to_json
           end
 
+          app.get '/ddtf/tests/new' do
+            halt 400, {
+                status: 'error',
+                data: nil,
+                message: 'Please provide stashTSDLink url'
+            }.to_json unless params[:stashTSDLink]
+
+            tsd = nil
+            begin
+              tsd_content = TsdUtils::ContentFetcher.load(params[:stashTSDLink])
+              tsd = JSON.parse(tsd_content.encode('ASCII', undef: :replace, replace: ''))
```

Comment:
Just a stylistic thing, but maybe Encoding:US-ASCII instead of string 'ASCII' would be less hacky?